PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/common/HDInsightHelperImpl.java

```diff
@@ -79,47 +78,53 @@ public String getInstallationId() {
         }
     }
 
-    public void openJobViewEditor(Object projectObject, String uuid) {
-        IClusterDetail clusterDetail = JobViewManager.getCluster(uuid);
+    public void openJobViewEditor(final Object projectObject, final String uuid) {
+        final IClusterDetail clusterDetail = JobViewManager.getCluster(uuid);
 
-        Project project = (Project)projectObject;
-        VirtualFile openedFile = getOpenedItem(project);
-        if(openedFile == null || isNeedReopen(openedFile, clusterDetail)) {
-            openItem(project,clusterDetail, uuid, openedFile);
+        final Project project = (Project) projectObject;
+        final VirtualFile openedFile = getOpenedItem(project);
+
+        // TODO: Fix the issue of clusterDetail may be null
+        if (openedFile == null || isNeedReopen(openedFile, clusterDetail)) {
+            openItem(project, clusterDetail, uuid, openedFile);
         } else {
             openItem(project, openedFile, null);
         }
     }
 
-    private boolean isNeedReopen(@NotNull VirtualFile virtualFile, @NotNull IClusterDetail myClusterDetail) {
-        IClusterDetail detail = virtualFile.getUserData(JobViewEditorProvider.JOB_VIEW_KEY);
+    private boolean isNeedReopen(@NotNull final VirtualFile virtualFile,
+                                 @NotNull final IClusterDetail myClusterDetail) {
+        final IClusterDetail detail = virtualFile.getUserData(JobViewEditorProvider.JOB_VIEW_KEY);
         return detail != null && !detail.getName().equalsIgnoreCase(myClusterDetail.getName());
     }
 
-    private static VirtualFile getOpenedItem(Project project) {
-        FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
-        for (VirtualFile editedFile : fileEditorManager.getOpenFiles()) {
-            IClusterDetail detail = editedFile.getUserData(JobViewEditorProvider.JOB_VIEW_KEY);
+    @Nullable
+    private static VirtualFile getOpenedItem(final Project project) {
+        final FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
+        for (final VirtualFile editedFile : fileEditorManager.getOpenFiles()) {
+            final IClusterDetail detail = editedFile.getUserData(JobViewEditorProvider.JOB_VIEW_KEY);
             if (detail != null) {
                 return editedFile;
             }
         }
         return null;
     }
 
-    private void openItem(@NotNull final Project project, @NotNull final VirtualFile virtualFile, @Nullable final VirtualFile closeableVirtualFile) {
-        ApplicationManager.getApplication().invokeLater(new Runnable() {
-            @Override
-            public void run() {
-                if(closeableVirtualFile != null) {
-                    FileEditorManager.getInstance(project).closeFile(closeableVirtualFile);
-                }
-                FileEditorManager.getInstance(project).openFile((VirtualFile) virtualFile, true, false);
+    private void openItem(@NotNull final Project project,
```

Comment:
Maybe we can remove all the @NotNull anotation in this file.

Reply:
It's out of coding style issue scope...We can fix them when touching these codes in development.

Reply:
But seems like you add some `Nullable` annotation to the existing code like the following one.
```java
    @Nullable
    private static VirtualFile getOpenedItem(final Project project) {
        final FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
        for (final VirtualFile editedFile : fileEditorManager.getOpenFiles()) {
            final IClusterDetail detail = editedFile.getUserData(JobViewEditorProvider.JOB_VIEW_KEY);
```

Reply:
Oh, no...

Reply:
Reverted.